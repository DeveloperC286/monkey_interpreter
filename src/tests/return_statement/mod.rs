use super::*;

#[template]
#[rstest(
    code,
    snapshot_name,
    case("return 10;", "return_statement_case_1"),
    case("return true;", "return_statement_case_2"),
    case("return \"string123\";", "return_statement_case_3")
)]
fn return_statement_cases(code: &str, snapshot_name: &str) {}

#[apply(return_statement_cases)]
fn test_return_statement_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(return_statement_cases)]
fn test_return_statement_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(return_statement_cases)]
fn test_return_statement_evaluation(code: &str, snapshot_name: &str) {
    assert_evaluation!(code, snapshot_name);
}

#[apply(return_statement_cases)]
fn test_return_statement_environment(code: &str, snapshot_name: &str) {
    assert_environment!(code, snapshot_name);
}
