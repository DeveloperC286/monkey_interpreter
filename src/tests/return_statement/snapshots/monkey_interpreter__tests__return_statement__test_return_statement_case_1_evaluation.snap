---
source: src/tests/return_statement/mod.rs
expression: evaluation
---
Integer {
    value: 10,
}
